File: brief/brief-springboot/src/main/java/cn/followtry/springboot/controller/TestController.java
/**
 * 
 */
package cn.followtry.springboot.controller;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import cn.followtry.springboot.springinaction.aop.Performance;

/**
 * @author jingzz
 * @time 2016年10月13日 上午9:09:48
 * @name brief-springboot/cn.followtry.springboot.controller.TestController
 * @since 2016年10月13日 上午9:09:48
 */
@RestController
public class TestController {
	
	@Autowired
	private Performance performance;
	
	@RequestMapping("/test")
	public void test(@RequestParam("name")String name){
		System.out.println("TestController.test()");
		performance.perform();
		System.out.println();
		System.out.println();
		System.out.println("TestController.test()2");
		performance.perform(name);
		System.out.println("TestController.test()3");
	}
}

File: brief-springboot-kotlin/src/main/kotlin/cn/followtry/web/TestController.kt
package cn.followtry.web

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

/**
 *
 * @author followtry
 * @since  2021/7/20 6:16 下午
 */
@RestController
@RequestMapping(value = ["ws/kotlin/web"])
open class TestController {

    @GetMapping(value = ["hello"])
    fun hello(name: String) : String {
        return "hello" + name
    }
}